### ISSUE TITLE
Alerts On Syslog does not provide option to send syslog to remote collector

### ISSUE THREAD

**Issue by djha983** [ntop/ntopng#1658]
Hi,

I see option in ntopng "Alerts On Syslog" which writes alerts to syslog on linux.
But in case of windows i don't see it writing anywhere in eventlog or in file.
Is Alerts On Syslog deature applicable in windows version of ntopng? is there any other configuration required for this?

Thanks,
Deepak jha

**Comment by lucaderi** [comment]
Windows has no concept of syslog so it can't be immediately supported there. Traces are written on the Windows system events that could also be used for this purpose. Are they enough for what you have in mind to do?

**Comment by djha983** [comment]
Hi,

Actually here is what I am trying to achieve.
•	I have installed ntopng on centos and I have enabled default alerts.
•	I have enabled feature “Alerts On Syslog”, So it writes alerts to Linux system log whenever alert is generated and same is forwarded to remote syslog receiver(EventTracker) by adding remote hostname in rsyslog.conf file.
•	Above works fine.
 
Here is the problem what I am facing:-
 
•	I have installed ntopng on windows.
•	I have enabled default alerts in ntopng and whenever alert condition matches ntopng alerts are visible in ntopng Web UI.
•	Now next thing what I want is I want those alerts to bring to SIEM(EventTracker),I can achieve this by reading event log on windows or from the logfile.
•	I want to know whether ntopng will also write alerts it to eventlog  or in any file.
•	I have also installed ntopng service by adding -T=3.But I don’t see any log file written in ntopng install directory.
 
Let me know, is it possible to achieve above in ntopng on windows? If yes how we can achieve.

I don’t find alerts_v7.db database in ntopng install path  .
What configuration I need to make in ntopng to achieve that?  Currently I am running ntopng service with below parameters.
ntopng /I -F=es;flows;ntopng-%Y.%m.%d;http://172.30.5.41:9200/_bulk; -T=3
 
Thanks

**Comment by simonemainardi** [comment]
> Actually here is what I am trying to achieve.
> •	I have installed ntopng on centos and I have enabled default alerts.
> •	I have enabled feature “Alerts On Syslog”, So it writes alerts to Linux system log whenever alert is generated and same is forwarded to remote syslog receiver(EventTracker) by adding remote hostname in rsyslog.conf file.
> •	Above works fine.

correct, configuring the rsyslog.conf is the proper way to instruct syslog to deliver events to remote syslog collectors.


> •	Now next thing what I want is I want those alerts to bring to SIEM(EventTracker),I can achieve this by reading event log on windows or from the logfile.
> •	I want to know whether ntopng will also write alerts it to eventlog or in any file.

On windows there's no concept of syslog and alerts are not written to file. For this reason, in order to retrieve the exising alerts under Windows, the recommended way is to query endpoint `get_alerts_data.lua` to have an easy-to-parse JSON representation of the alerts detected:

```
http://devel:3000/lua/get_alerts_data.lua?status=historical&currentPage=1&perPage=10&sortColumn=column_date&sortOrder=desc
```

The most important parameter you can control is the `status`, that allows you to chose between currently `engage`d alerts, past `historical` alerts and `historical-flows` alerts.

So basically if you want to retrieve engaged alerts, all you have to do is to query as

```
http://devel:3000/lua/get_alerts_data.lua?status=engaged&currentPage=1&perPage=10&sortColumn=column_date&sortOrder=desc
```

The other parameters are pretty intuitive and are used to paginate and sort the result set.

The endpoint mentioned above is the one used by the web UI to populate the alerts page

![image](https://user-images.githubusercontent.com/3397663/44522288-5095b200-a6d6-11e8-85aa-54767c83a0da.png)

So you can be sure you'll always get all the results.
